base/array.jl

```diff
@@ -15,7 +15,8 @@ typealias StridedVector{T,A<:DenseArray,I<:Tuple{Vararg{RangeIndex}}}  Union(Den
 typealias StridedMatrix{T,A<:DenseArray,I<:Tuple{Vararg{RangeIndex}}}  Union(DenseArray{T,2}, SubArray{T,2,A,I})
 typealias StridedVecOrMat{T} Union(StridedVector{T}, StridedMatrix{T})
 
-call{T}(::Type{Vector{T}}, m::Integer) = Array{T}(m)
+call{T}(::Type{Vector{T}}, m::Integer=0) = Array{T}(m)
```

Comment:
Hmm, in an [earlier revision](https://github.com/garrison/julia/commit/b265bb07c0d5f8310da91fd4f565b15a66761013) I added precisely this line instead of doing the current replacement.  Why do you prefer it?